src/App.js

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import './App.css';
+import Header from './components/Header';
+import PokemonTiles from './components/PokemonTiles';
+import apiRequest from "./RestApiRequest";
 
 function App() {
-  return (
-    <div>
-      Begin hier met de opdracht!
-    </div>
-  );
+
+    const [data, setData] = useState( [] );
+
+    useEffect( ()=> {
+        const source = apiRequest( "https://pokeapi.co/api/v2/pokemon", setData );
+
+        return function cleanup() {
+            source.cancel();
+        }
+    }, []);
+
+    return (
+        <div>
+            <Header data={data} setData={setData}/>
+            {data.results &&
+                <PokemonTiles results={data.results} data={data}/>
+            }
```

Comment:
Hier moet je dan wel nog een check toevoegen, want in beginsel is `data` undefined en krijg je dus errors op `.results` bijvoorbeeld.
```suggestion
            {data && (
                <>
                    <Header data={data} setData={setData}/>
                    {data.results &&
                        <PokemonTiles results={data.results} data={data}/>
                    }
                </>
            )}
```